CHANGELOG.rst

```diff
@@ -66,6 +66,8 @@ Changed
   is not longer a comma-separated string but a list of strings. For example,
   this basis ``['u1','u2','u3','cx']`` should be used instead of ``'u1,u2,u3,cx'``
   (#1333)
+- Methods on the ``DAGCircuit`` which previously returned node_ids and/or dicts now
+  return ``DAGNodes``
```

Comment:
On second though, since we're also changing many of the names of the dag methods for the 0.8 release, might the overall deprecation process be simpler if we deprecate but do not modify the existing `get_*` methods and update only their replacement methods to take and return `DAGNodes`? 

This would also resolve the breaking change around the change of return type for these methods, e.g. code that calls `dag.get_op_nodes()` and expects a list of integer node ids can continue to do so until it is transitions to `dag.op_nodes` where it will expect `DAGNode`s.

Reply:
I have done this for the `get_*` methods. However there are still some breaking changes around methods such as `successors()` which has not has its name changed but now returns a list of `DAGNode`s instead of `node_id`s